public_html/layouts/resources/Fields.js

```diff
@@ -697,6 +707,50 @@ App.Fields = {
 				}
 			});
 		},
+		registerCustomFilterOptionsHoverEvent(element) {
+			element.on('mouseenter mouseleave', 'li.select2-results__option[role="treeitem"]', (event) => {
+				let liElement = $(event.currentTarget),
+					liUserOptions = liElement.find('.js-user-actions');
+				if (liElement.hasClass('group-result')) {
+					return;
+				}
+				if (event.type === 'mouseenter' && liUserOptions.length === 0) {
+					this.appendOptionActionsTemplate(liElement);
+				}
+			});
+		},
+		userOptionClickEvent(element) {
+			element.on('mouseup', '[data-fa-i2svg].js-user-favorites', function (event) {
+				let thisInstance = $(this),
+					liElement = thisInstance.closest('.select2-results__option'),
+					id = liElement.attr("id"),
+					idArr = id.split("-"),
+					currentOptionId = '';
+				if (idArr.length > 0) {
+					currentOptionId = idArr[idArr.length - 1];
+				}
+				let params = {
+					userId: currentOptionId
+				};
+				//AppConnector.request(params).done(function (data) {
+				if (thisInstance.data('prefix') === 'far') {
+					thisInstance.addClass('fas').removeClass('far');
+					thisInstance.data('prefix', 'fas');
+				} else {
+					thisInstance.addClass('far').removeClass('fas');
+					thisInstance.data('prefix', 'far');
+				}
+				//});
+				event.stopPropagation();
+			});
+		},
+		appendOptionActionsTemplate(liElement) {
+			let template = $(`<span class="js-user-actions o-filter-actions noWrap float-right">
```

Comment:
change js-user-actions, data-value="favorites", js-user-favorites to more general.
get icons from .data() elements